tiledb/common/thread_pool/thread_pool.cc

```diff
@@ -30,28 +30,34 @@
  * This file defines the ThreadPool class.
  */
 
-#ifdef LEGACY_THREAD_POOL
-#include "legacy_thread_pool.cc"
-#else
-
 #include <cassert>
 #include <memory>
 #include <queue>
+#include <thread>
 
 #include "tiledb/common/logger.h"
 #include "tiledb/common/thread_pool.h"
 
 namespace tiledb::common {
 
-Status ThreadPool::init(size_t n) {
-  if (n == 0) {
-    return Status_ThreadPoolError(
-        "Unable to initialize a thread pool with a concurrency level of 0.");
+ThreadPool::ThreadPool(size_t n)
+    : concurrency_level_(n) {
+  // If concurrency_level_ is set to zero, shut down the thread pool
+  if (concurrency_level_ == 0) {
+    task_queue_.drain();
```

Comment:
Since we are in the constructor how/why would there be any tasks to drain here?

Reply:
The `concurrency_level_ == 0` is a special condition in the constructor that constructs the `ThreadPool` in its shutdown state.  The `drain()` function puts the queue into its shutdown state.